### PR TITLE
feat(ingestion/powerbi): support multiple tables as upstream in native SQL parsing

### PR DESCRIPTION
**Pull request by siddiquebagwan-gslab**

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/native_sql_parser.py [hunk 1]
```diff
@@ -45,3 +52,48 @@ def get_tables(native_query: str) -> List[str]:
...
+) -> Optional["SqlParsingResult"]:
+
+    logger.debug("Using sqlglot_lineage to parse custom sql")
+
+    sql_query = remove_special_characters(query)
+    logger.debug(f"Parsing sql={sql_query}")
+
+    parsed_result: Optional["SqlParsingResult"] = None
+    try:
+        schema_resolver = (
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/native_sql_parser.py:74]: some of this logic looks like it was copy-pasted from elsewhere

can we instead extract it into a common helper method?

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/resolver.py [hunk 2]
```diff
@@ -352,7 +460,11 @@ def resolve_to_data_platform_table_list(self) -> List[DataPlatformTable]:
                 continue
 
             table_full_name_creator: AbstractDataPlatformTableCreator = (
-                supported_resolver.get_table_full_name_creator()()
+                supported_resolver.get_table_full_name_creator()(
+                    ctx=ctx,
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/resolver.py:464]: eventually, will it be possible to get rid of this `AbstractDataPlatformTableCreator` / move some of that logic into the sqlglot_lineage parser itself?

Overall I found it pretty tricky to follow this code and understand exactly what it was doing

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/native_sql_parser.py (hunk 1 above)
Comment (siddiquebagwan-gslab) [reply metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/native_sql_parser.py:74]: I added it in sqlglot_lineage as one additional method

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/resolver.py (hunk 2 above)
Comment (siddiquebagwan-gslab) [reply metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/resolver.py:464]: AbstractDataPlatformTableCreator is an abstraction over M-Query parsing and different classes are extending it to support special construct of M-Query, like RedShift, Snowflake, Postgres, MSSQL and some of them has native query inside M-Query code. In current code SNOWFLAKE and AMAZON_REDSHIFT supports M-Query way to create transformation as well as custom sql to create transformation inside M-Query. Both are using `AbstractDataPlatformTableCreator.parse_custom_sql` to generate lineage if during M-Query parsing they found custom query inside M-Query

File: metadata-ingestion/src/datahub/utilities/sqlglot_lineage.py [hunk 3]
```diff
@@ -825,3 +825,43 @@ def sqlglot_lineage(
                 table_error=e,
             ),
         )
+
+
+def create_lineage_sql_parsed_result(
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/utilities/sqlglot_lineage.py:830]: please also refactor tableau to use this method

File: metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/resolver.py (hunk 2 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/resolver.py:464]: Please add some comments to this file about how it is structured at a high level and why it was done that way

Comment (siddiquebagwan-gslab) [reply metadata-ingestion/src/datahub/ingestion/source/powerbi/m_query/resolver.py:464]: done

File: metadata-ingestion/src/datahub/utilities/sqlglot_lineage.py (hunk 3 above)
Comment (siddiquebagwan-gslab) [reply metadata-ingestion/src/datahub/utilities/sqlglot_lineage.py:830]: done

**Review by hsheth2 — APPROVED** [review]

**Comment by anshbansal** [comment]
Failure not related to ingestion code so merging

**Pull request merged**